src/UnitTests/XpressionMapperTests.cs

```diff
@@ -147,6 +147,39 @@ public void Map__object_including_child_and_grandchild()
 
             //Assert
             Assert.True(users.Count == 2);
+        }
+
+        [Fact]
+        public void Map_object_when_null_values_are_typed()
+        {
+            //Arrange
+            //Expression<Func<UserModel, bool>> selection = s => s != null && s.AccountModel != null && s.AccountModel.Bal > 555.20;

```

Comment:
Casting nulls doesn't do the job?

    Expression<Func<UserModel, bool>> selection = s => s != (UserModel) null && s.AccountModel != (AccountModel) null && s.AccountModel.Bal > 555.20;